indexer/services/comlink/src/controllers/api/v4/affiliates-controller.ts

```diff
@@ -0,0 +1,72 @@
+import { stats } from '@dydxprotocol-indexer/base';
+import express from 'express';
+import { checkSchema, matchedData } from 'express-validator';
+import {
+  Controller, Get, Query, Route,
+} from 'tsoa';
+
+import { getReqRateLimiter } from '../../../caches/rate-limiters';
+import config from '../../../config';
+import { handleControllerError } from '../../../lib/helpers';
+import { rateLimiterMiddleware } from '../../../lib/rate-limit';
+import ExportResponseCodeStats from '../../../request-helpers/export-response-code-stats';
+import { AffiliateReferralCodeRequest, AffiliateReferralCodeResponse } from '../../../types';
+
+const router: express.Router = express.Router();
+const controllerName: string = 'affiliates-controller';
+
+@Route('affiliates')
+class AffiliatesController extends Controller {
+  @Get('/referral_code')
+  async getReferralCode(
+    @Query() address: string,
+  ): Promise<AffiliateReferralCodeResponse> {
+
+    // simulate a delay
+    await new Promise((resolve) => setTimeout(resolve, 100));
+
+    return {
+      referralCode: `${address}-referral_code`,
```

Comment:
Lets simulate what an actual referral code would look like - can you truncate to 13 characters which would be the max length here? Address would be too long.

For context the referral code is going to be (Adjective + noun + 3 digit number)